### ISSUE TITLE
@casl/mongoose should use `.and` for adding the condition instead of `.where` when called on a Query

### ISSUE THREAD

**Issue by m-weeks** [stalniy/casl#272]
I discovered this issue when I was writing a query that had an `$and` condition, and my query log was showing the condition as `$and: [{}]`

Given the following example:

```
appointments = await Appointment.find({
  $and: [
    { startDate: { $gte: new Date('2020-01-01') } },
    { startDate: { $lte: new Date('2020-01-05') } },
  ]
}).accessibleBy(ability)
```

The problem is because the `accessibleBy` method uses `.where` which will override the `$and` parameter, where as the `.and` method will add a new condition.

**Comment by stalniy** [comment]
Thanks for the issue! Could you please tell which mongoose version do you use?

**Comment by stalniy** [comment]
I’ll look into this tomorrow morning by Kyiv time. As a workaround I suggest you merge your `$and` into single object:

```js
appointments = await Appointment.find({
    startDate: { 
        $gte: new Date('2020-01-01'),
        $lte: new Date('2020-01-05') 
    },
}).accessibleBy(ability)
```

**Comment by m-weeks** [comment]
@stalniy I am currently on mongoose 5.9.2

I created a PR that I believe solves the issue if you wanted to take a look at it

I am currently combining my conditions, but it would be ideal to have it fixed as it makes it trickier to dynamically generate query conditions, especially if you're adding multiple conditions on the same value.

**Comment by stalniy** [comment]
Sure, I believe I’ll merge in few hours. Just need some time to make sure everything is ok

**Comment by stalniy** [comment]
Released in `@casl/mongoose@2.3.3`
